dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,402 @@
+            db_type = "DateTime"
+        if datetime_match := re.match(
+            r"DateTime64(?:\((?P<precision>\d+)(?:,?\s*'(?P<timezone>UTC)')?\))?", db_type
+        ):
+            if datetime_match["precision"]:
+                precision = int(datetime_match["precision"])
+            else:
+                precision = None
+            db_type = "DateTime64"
+
+        # Extract precision and scale, parameters and remove from string.
+        if decimal_match := re.match(
+            r"Decimal\((?P<precision>\d+)\s*(?:,\s*(?P<scale>\d+))?\)", db_type
+        ):
+            precision, scale = decimal_match.groups()  # type: ignore[assignment]
+            precision = int(precision)
+            scale = int(scale) if scale else 0
+            db_type = "Decimal"
+
+        if db_type == "Decimal" and (precision, scale) == self.capabilities.wei_precision:
+            return dict(data_type="wei")
+
+        return super().from_db_type(db_type, precision, scale)
+
+
+class ClickHouseLoadJob(LoadJob, FollowupJob):
+    def __init__(
+        self,
+        file_path: str,
+        table_name: str,
+        client: ClickHouseSqlClient,
+        staging_credentials: Optional[CredentialsConfiguration] = None,
+    ) -> None:
+        file_name = FileStorage.get_file_name_from_file_path(file_path)
+        super().__init__(file_name)
+
+        qualified_table_name = client.make_qualified_table_name(table_name)
+        bucket_path = None
+
+        if NewReferenceJob.is_reference_job(file_path):
+            bucket_path = NewReferenceJob.resolve_reference(file_path)
+            file_name = FileStorage.get_file_name_from_file_path(bucket_path)
+            bucket_url = urlparse(bucket_path)
+            bucket_scheme = bucket_url.scheme
+
+        ext = cast(SUPPORTED_FILE_FORMATS, os.path.splitext(file_name)[1][1:].lower())
+        clickhouse_format: str = FILE_FORMAT_TO_TABLE_FUNCTION_MAPPING[ext]
+
+        compression = "auto"
+
+        # Don't use dbapi driver for local files.
+        if not bucket_path:
+            # Local filesystem.
+            if ext == "jsonl":
+                compression = "gz" if FileStorage.is_gzipped(file_path) else "none"
+            try:
+                with clickhouse_connect.create_client(
+                    host=client.credentials.host,
+                    port=client.credentials.http_port,
+                    database=client.credentials.database,
+                    user_name=client.credentials.username,
+                    password=client.credentials.password,
+                    secure=bool(client.credentials.secure),
+                ) as clickhouse_connect_client:
+                    insert_file(
+                        clickhouse_connect_client,
+                        qualified_table_name,
+                        file_path,
+                        fmt=clickhouse_format,
+                        settings={
+                            "allow_experimental_lightweight_delete": 1,
```

Comment:
I use the lightweight delete in the merge job, it seems to be working well.

Reply:
I think this is the only way to go :)